frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/SquashCurrentAction.java

```diff
@@ -26,17 +26,18 @@ protected void onUpdate(AnActionEvent anActionEvent) {
       return;
     }
 
-    val branchNameOption = getNameOfBranchUnderAction(anActionEvent);
-    val nonRootBranchOption = branchNameOption.flatMap(bn -> getManagedBranchByName(anActionEvent, bn))
-        .flatMap(b -> b.isNonRoot() ? Option.of(b.asNonRoot()) : Option.none());
-    val syncToParentStatus = nonRootBranchOption.map(b -> b.getSyncToParentStatus()).getOrNull();
-    val numberOfCommits = nonRootBranchOption.map(b -> b.getCommits().length()).getOrNull();
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val managedBranch = getManagedBranchByName(anActionEvent, branchName);
+    val nonRootBranch = managedBranch != null && managedBranch.isNonRoot()
+        ? managedBranch.asNonRoot()
+        : null;
+    val syncToParentStatus = nonRootBranch != null ? nonRootBranch.getSyncToParentStatus() : null;
 
-    val branchName = branchNameOption.getOrNull();
-    if (branchName != null && nonRootBranchOption.isEmpty()) {
+    if (branchName != null && nonRootBranch == null) {
       presentation.setVisible(false);
 
-    } else if (branchName != null && numberOfCommits != null) {
+    } else if (branchName != null && nonRootBranch != null) { //nonRootBranch is certainly not null.
```

Comment:
+1, delete comment, or just reorganize the ifs in a more Checker-friendly way (with `if (branchName != null) {` as the top-level condition)

Reply:
yes it is like the other situation.